TELCODOCS-778 OVN/OVS Troubleshooting guide (#54155)
openshift/openshift-docs: networking/ovn_kubernetes_network_provider/ovn-kubernetes-tracing-using-ovntrace.adoc

:_content-type: ASSEMBLY
[id="ovn-kubernetes-tracing-with-ovnkube"]
= Tracing Openflow with ovnkube-trace
include::_attributes/common-attributes.adoc[]
:context: ovn-kubernetes-tracing-with-ovnkube

toc::[]

> OVN and OVS traffic flows can be simulated in a single utility called `ovnkube-trace`. The `ovnkube-trace` utility runs `ovn-trace`, `ovs-appctl ofproto/trace` and `ovn-detrace` and correlates all that information in one easy to read output.

Reviewer: ```suggestion
OVN and OVS traffic flows can be simulated in a single utility called `ovnkube-trace`. The `ovnkube-trace` utility runs `ovn-trace`, `ovs-appctl ofproto/trace` and `ovn-detrace` and correlates that information in a single output.
```

> The `ovnkube-trace` binary can be executed from a dedicated container, or since {product-title} 4.7, the binary can be copied to a local host and executed from there.

Reviewer: ```suggestion
You can execute the `ovnkube-trace` binary from a dedicated container. For releases after {product-title} 4.7, you can also copy the binary to a local host and execute it from that host.
```

Reviewer: suggestion: avoid passive voice

[NOTE]
====

> The binary in the quay images do not currently work for Dual IP stack or IPv6 only environments, for those environments it is required to build from source.

Reviewer: ```suggestion
The binaries in the Quay images do not currently work for Dual IP stack or IPv6 only environments. For those environments, you must build from source.
```

====

> include::modules/nw-ovn-kubernetes-install-ovnkube-trace-local.adoc[leveloffset=+1]

Reviewer: module reviewed by Max

> include::modules/nw-ovn-kubernetes-running-ovnkube-trace.adoc[leveloffset=+1]

Reviewer: reviewed by Max
